core/commands/dag/dag.go

```diff
@@ -241,3 +261,249 @@ var DagResolveCmd = &cmds.Command{
 	},
 	Type: ResolveOutput{},
 }
+
+type importResult struct {
+	roots map[cid.Cid]bool
+	err   error
+}
+
+var DagImportCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Import the contents of .car files",
+		ShortDescription: `
+'ipfs dag import' parses .car files and adds all objects to the blockstore.
+Currently only car files with version 1 are supported, without particular
+hard limits on the contents (so called car-determinism is not enforced).
+
+By default, after all supplied (one or more) car files have been processed,
+an attempt is made to pin recursively each individual root specified in any car
+headers. This is done at the end of the import, allowing pinning to take place
+over the contents of all cars (a dag scattered over multiple files) plus
+anything present in the blockstore.
```

Comment:
I'd rewrite this to focus on _what_ happens, not how, and to remove terms like "car-determinism" (not something anyone outside of the core IPLD team will understand).

Something along the lines of:

> `ipfs dag import` imports all blocks present in the CAR (Content Addressed aRchive) file, pinning the roots (as specified in the CAR file headers) unless `--pin-roots` is set to false.
>
> Note:
> 1. This command will add all blocks in the CAR file, not just those reachable from the roots. However, these other blocks will not be pinned and may be garbage collected later.
> 2. other caveats...
> 
> Maximum supported CAR version: 1

Reply:
Done
Result:
```
SYNOPSIS
  ipfs dag import [--silent] [--pin-roots=false] [--] <path>...

ARGUMENTS

  <path>... - The path of a .car file.

OPTIONS

  --silent     bool - No output.
  --pin-roots  bool - Pin optional roots listed in the .car headers after importing. Default: true.

DESCRIPTION

  'ipfs dag import' imports all blocks present in supplied .car
  ( Content Address aRchive ) files, recursively pinning any roots
  specified in the CAR file headers, unless --pin-roots is set to false.
  
  Note:
  	This command will import all blocks in the CAR file, not just those
  	reachable from the specified roots. However, these other blocks will
  	not be pinned and may be garbage collected later.
  
  	The pinning of the roots happens after all car files are processed,
  	permitting import of DAGs spanning multiple files.
  
  	Pinning takes place in offline-mode exclusively, with no option to change
  	this: if the collection of blocks from the imported CAR files and what
  	is currently present in the blockstore is insufficient pinning of the
  	affected root will fail
  
  Maximum supported CAR version: 1
```